src/machine/lib_machine.rs

```diff
@@ -18,9 +18,9 @@ use super::{
 
 pub struct QueryState<'a> {
     machine: &'a mut Machine,
-    term: TermWriteResult,
+    pub(crate) term: TermWriteResult,
     stub_b: usize,
-    var_names: IndexMap<HeapCellValue, VarPtr>,
+    pub(crate) var_names: IndexMap<HeapCellValue, VarPtr>,
     called: bool,
```

Comment:
I don't see new usages of these fields, therefor I think it's no longer necessary to change the visibility of these fields.